src/components/DateInput.jsx

```diff
@@ -217,7 +220,7 @@ class DateInput extends React.PureComponent {
             focused && styles.DateInput_input__focused,
             disabled && styles.DateInput_input__disabled,
           )}
-          aria-label={placeholder}
+          aria-label={noAriaLabel ? '' : placeholder}
```

Comment:
This would result in the label for the node to be the empty string instead of not setting it. What you'd want to do instead is set the prop to be undefined or null if you don't want it to be set.